File: vSongApp/src/main/java/com/jackson_siro/visongbook/models/CountryModel.java
package com.jackson_siro.visongbook.models;

import com.jackson_siro.visongbook.core.Searchable;

public class CountryModel implements Searchable {
    private String Country;
    private String CountryCode;
    private String IsoCode;

    public CountryModel(String country, String countryCode, String isoCode) {
        Country = country;
        CountryCode = countryCode;
        IsoCode = isoCode;
    }

    @Override
    public String getTitle() {
        return getCcode() + " | " + getCountry() + " - " + getIcode();
    }

    public String getCountry() {
        return Country;
    }

    public CountryModel setName(String country) {
        Country = country;
        return this;
    }

    public String getCcode() {
        return CountryCode;
    }

    public CountryModel setShort(String countryCode) {
        CountryCode = countryCode;
        return this;
    }

    public String getIcode() {
        return IsoCode;
    }

    public CountryModel setCode(String isoCode) {
        IsoCode = isoCode;
        return this;
    }
}

File: vSongApp/src/main/java/com/jackson_siro/visongbook/models/CountryModel.kt
package com.jackson_siro.visongbook.models

import com.jackson_siro.visongbook.core.Searchable

class CountryModel(country: String, countryCode: String, isoCode: String) : Searchable {
    var country: String? = null
        private set
    var ccode: String? = null
        private set
    var icode: String? = null
        private set

    init {
        this.country = country
        ccode = countryCode
        icode = isoCode
    }

    override fun getTitle(): String {
        return "$ccode | $country - $icode"
    }

    fun setName(country: String): CountryModel {
        this.country = country
        return this
    }

    fun setShort(countryCode: String): CountryModel {
        ccode = countryCode
        return this
    }

    fun setCode(isoCode: String): CountryModel {
        icode = isoCode
        return this
    }
}
